source/ui/resolution.js

```diff
@@ -0,0 +1,194 @@
+		_screenTypeObject;
+
+	var getScreenTypeObject = function (name) {
+		if (name == _screenType && _screenTypeObject) {
+			return _screenTypeObject;
+		}
+		var types = _screenTypes;
+		return types.filter(function (elem) {
+			return (name == elem.name);
+		})[0];
+	};
+
+	/**
+	* @namespace enyo.ri
+	*/
+	enyo.ri = {
+		/**
+		* Setup screen resolution scaling capabilities by defining all of the screens you're working
+		* with. These should be in the order of smallest to largest (according to width). Running
+		* this also initializes the rest of this resolution code.
+		*
+		* In the arguments, the following properties are required: 'name', 'pxPerRem', 'width'.
+		* The property 'base' defines the primary or default resoultion that everything else will be
+		* based upon.
+		*
+		* ```
+		* enyo.ri.defineScreenTypes([
+		* 	{name: 'hd',    pxPerRem: 16, height: 720,  width: 1280},
+		* 	{name: 'fhd',   pxPerRem: 24, height: 1080, width: 1920, base: true},
+		* 	{name: 'uhd',   pxPerRem: 48, height: 2160, width: 3840}
+		* ]);
+		* ```
+		*
+		* @param {Array} types An array of objects with arguments like the example
+		* @public
+		*/
+		defineScreenTypes: function (types) {
+			_screenTypes = types;
+			for (var i = 0; i < _screenTypes.length; i++) {
+				if (_screenTypes[i]['base']) _baseScreenType = _screenTypes[i].name;
+			}
+			enyo.ri.init();
+		},
+
+		/**
+		* Fetches the best-matching screen type name for the current screen size. The "best" screen type
+		* is determined by the screen type name that is the closest to the screen resolution without
+		* going over. ("The Price is Right" style.)
+		*
+		* @param {Object} [rez] - Optional measurement scheme. Must have "height" and "width" properties.
+		* @returns {String} Screen type, like "fhd", "uhd", etc.
+		* @public
+		*/
+		getScreenType: function (rez) {
+			rez = rez || {
+				height: scope.innerHeight,
+				width: scope.innerWidth
+			};
+			var i,
+				types = _screenTypes,
+				bestMatch = types[types.length - 1].name;
+
+			// loop thorugh resolutions
+			for (i = types.length - 1; i >= 0; i--) {
+				// find the one that matches our current size or is smaller. default to the first.
+				if (rez.width <= types[i].width) {
+					bestMatch = types[i].name;
+				}
+			}
+			// return the name of the resolution if we find one.
+			return bestMatch;
+		},
+
+		/**
+		* @private
+		*/
+		updateScreenTypeOnBody: function (type) {
+			type = type || _screenType;
+			if (type) {
+				enyo.dom.addBodyClass('enyo-res-' + type.toLowerCase());
+				return type;
+			}
+		},
+
+		/**
+		* @private
+		*/
+		getRiRatio: function (type) {
+			type = type || _screenType;
+			if (type) {
+				return this.getUnitToPixelFactors(type) / this.getUnitToPixelFactors(_baseScreenType);
+			}
+			return 1;
+		},
+
+		/**
+		* @private
+		*/
+		getUnitToPixelFactors: function (type) {
+			type = type || _screenType;
+			if (type) {
+				_riRatio = getScreenTypeObject(type).pxPerRem;
```

Comment:
This does seem to be a general problem that didn't reveal itself earlier. I'll look into it.